services/payments/src/simcore_service_payments/services/stripe.py

```diff
@@ -0,0 +1,96 @@
+""" Interface to communicate with the Resource Usage Tracker (RUT)
+
+- httpx client with base_url to PAYMENTS_RESOURCE_USAGE_TRACKER
+
+"""
+
+import contextlib
+import functools
+import logging
+from collections.abc import Callable
+
+import httpx
+from fastapi import FastAPI
+from httpx import HTTPStatusError
+from models_library.payments import StripeInvoiceID
+from servicelib.fastapi.app_state import SingletonInAppStateMixin
+from servicelib.fastapi.http_client import (
+    AttachLifespanMixin,
+    BaseHTTPApi,
+    HealthMixinMixin,
+)
+
+from ..core.errors import StripeRuntimeError
+from ..core.settings import ApplicationSettings
+from ..models.stripe import InvoiceData
+
+_logger = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def _raise_as_stripe_error():
+    """https://docs.stripe.com/api/errors"""
+    try:
+        yield
+
+    except HTTPStatusError as err:
+        raise StripeRuntimeError from err
```

Comment:
IMO this is still too little. Let's please follow up on this.

Reply:
Yes we will, I need to put some thinking into it, and let's discuss.